### ISSUE TITLE
list_immutable_data_history : Unexpected error using default options

### ISSUE THREAD

**Issue by larrysalibra** [stx-labs/stacks.js#154]
Using the default options (ie. pressing return) when running `list_immutable_data_history` in interactive mode result in an unexpected error (see output below). I'd expect that the default options would produce an error that the name doesn't exist and/or the immutable datum doesn't exist.

```
(venv) larry@blockstack-explorer-1:~$ blockstack list_immutable_data_history larry.id
[2016-10-27 11:17:32,830] [DEBUG] [spv:103] (8811.140172774102848) Using mainnet
[2016-10-27 11:17:33,071] [DEBUG] [cli:210] (8811.140172774102848) Enabling advanced methods
usage: blockstack list_immutable_data_history [-h] name data_id
blockstack list_immutable_data_history: error: too few arguments
[2016-10-27 11:17:33,135] [DEBUG] [client:134] (8811.140172774102848) Loaded storage driver 'disk'
[2016-10-27 11:17:33,136] [DEBUG] [client:134] (8811.140172774102848) Loaded storage driver 'blockstack_resolver'
[2016-10-27 11:17:33,136] [DEBUG] [client:134] (8811.140172774102848) Loaded storage driver 'blockstack_server'
[2016-10-27 11:17:33,137] [DEBUG] [client:134] (8811.140172774102848) Loaded storage driver 'http'
[2016-10-27 11:17:33,137] [DEBUG] [client:134] (8811.140172774102848) Loaded storage driver 'dht'

Interactive prompt engaged.  Press Ctrl+C to quit
Help for 'list_immutable_data_history': List all prior hashes of a given immutable datum

The name whose data to list ('name'): 
The data identifier whose history to list ('data_id'): 
[2016-10-27 11:17:48,360] [ERROR] [blockstack:80] (8811.140172774102848) 'unicode' object has no attribute 'has_key'
Traceback (most recent call last):
  File "/home/dev/explorer-api/venv/bin/blockstack", line 67, in <module>
    result = run_cli()
  File "/home/dev/explorer-api/venv/local/lib/python2.7/site-packages/blockstack_client/cli.py", line 287, in run_cli
    result = method( args, config_path=config_path )
  File "/home/dev/explorer-api/venv/local/lib/python2.7/site-packages/blockstack_client/actions.py", line 2182, in cli_advanced_list_immutable_data_history
    result = list_immutable_data_history(str(args.name), str(args.data_id))
  File "/home/dev/explorer-api/venv/local/lib/python2.7/site-packages/blockstack_client/data.py", line 349, in list_immutable_data_history
    zonefiles = list_zonefile_history( name, current_block=current_block, proxy=proxy )
  File "/home/dev/explorer-api/venv/local/lib/python2.7/site-packages/blockstack_client/data.py", line 322, in list_zonefile_history
    zonefile_hashes = list_update_history( name, current_block=current_block, proxy=proxy )
  File "/home/dev/explorer-api/venv/local/lib/python2.7/site-packages/blockstack_client/data.py", line 305, in list_update_history
    if history_data.has_key('value_hash') and history_data['value_hash'] is not None:
AttributeError: 'unicode' object has no attribute 'has_key'
{
    "error": "Unexpected error:\nTraceback (most recent call last):\n  File \"/home/dev/explorer-api/venv/bin/blockstack\", line 67, in <module>\n    result = run_cli()\n  File \"/home/dev/explorer-api/venv/local/lib/python2.7/site-packages/blockstack_client/cli.py\", line 287, in run_cli\n    result = method( args, config_path=config_path )\n  File \"/home/dev/explorer-api/venv/local/lib/python2.7/site-packages/blockstack_client/actions.py\", line 2182, in cli_advanced_list_immutable_data_history\n    result = list_immutable_data_history(str(args.name), str(args.data_id))\n  File \"/home/dev/explorer-api/venv/local/lib/python2.7/site-packages/blockstack_client/data.py\", line 349, in list_immutable_data_history\n    zonefiles = list_zonefile_history( name, current_block=current_block, proxy=proxy )\n  File \"/home/dev/explorer-api/venv/local/lib/python2.7/site-packages/blockstack_client/data.py\", line 322, in list_zonefile_history\n    zonefile_hashes = list_update_history( name, current_block=current_block, proxy=proxy )\n  File \"/home/dev/explorer-api/venv/local/lib/python2.7/site-packages/blockstack_client/data.py\", line 305, in list_update_history\n    if history_data.has_key('value_hash') and history_data['value_hash'] is not None:\nAttributeError: 'unicode' object has no attribute 'has_key'\n",
    "help": "Try getting latest version of CLI with \"sudo pip install blockstack --upgrade\""
}
(venv) larry@blockstack-explorer-1:~$

```

**Comment by jcnelson** [comment]
Thanks, will send a patch.  Just a heads-up: advanced methods aren't nearly as well tested in the CLI as the basic methods.  We test the library calls that make up the advanced methods, but the code that parses advanced method CLI arguments and feeds them into the library calls isn't that well tread.  I'm trying to extend the integration test suite to run shell commands instead of library calls to rectify this.

**Comment by jcnelson** [comment]
Pushed a possible fix for this particular problem in cd2c44ccf09374e00dbe5f496069d16ab345b205.  I think `get_name_blockchain_history` had failed earlier in the call (but that's not good either).  The patch just makes sure to print the error message, so we can find out why.
